src/main/java/de/uni_tuebingen/ub/nppm/servlet/backend/AdministrationBaumstrukturServlet.java

```diff
@@ -18,10 +18,18 @@ protected String getTitle() {
 
     @Override
     protected void generatePage(HttpServletRequest request, HttpServletResponse response) throws Exception {
-        RequestDispatcher rd = request.getRequestDispatcher("admin.baumstruktur.jsp");
-        rd.include(request, response);
+         //Use isAdminRequired to check if the user is an administrator.
+        if (isAdminRequired(request, response)) {
+            RequestDispatcher rd = request.getRequestDispatcher("admin.baumstruktur.jsp");
+            rd.include(request, response);
+        }else {
+            //If the user is not an administrator, redirect them to logout.jsp.
+            RequestDispatcher dispatcher = request.getRequestDispatcher("/logout.jsp");
```

Comment:
This check will be done by the parent function. The generatePage function should stay as before, you should only override the isAdminRequired function.